envoy/type/v2/range.proto

```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+// [#proto-status: draft]
+
+package envoy.type.v2;
+option go_package = "v2";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Range]
+
+// Specifies the signed int64 start and end of the range using half-open interval semantics [start,
+// end).
+message SInt64Range {
+  // start of the range (inclusive)
+  sint64 start = 1;
```

Comment:
Google APIs never use sint64, since negative values are rarely used in practice. You are paying more encoding/decoding cost to save wire format for negative value. At Google, even we process billions of requests per second, we sill favor simplicity than wire cost.